### PR TITLE
DATAUP-595 fix report not showing in app cell results tab

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

This was a little bit of drift from the last merge from `develop`. In the `truss` branch, we modified the `kbaseReportView` to not render until specifically told to. This helps with testing by managing Promises and data flow better. The `develop` branch got a recent update that doesn't do that, so this got a little bit lost.

Tests pending on the app cell results viewer tab

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

**Review by ialarmedalien — APPROVED** [review]
👍

File: kbase-extension/scss/partials/_appCell.scss [hunk 1]
```diff
@@ -446,3 +446,8 @@ Buttons
   font-style: italic;
   padding-left: 5px;
 }
+
+.kb-app-results-tab {
```
Comment (briehl) [inline kbase-extension/scss/partials/_appCell.scss:450]: moved inline CSS here.

File: kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js [hunk 2]
```diff
@@ -51,6 +51,21 @@ define([
...
+         *   report_name {string} name of the report object in the given workspace
+         *   report_ref {string} a report object reference of the format x/y/z. If present, overrides
+         *     the workspace_name and report_name options
+         *   showReportText {boolean} (default true) true if the report text should be shown
+         *   showCreatedObjects {boolean} (default false) if true, show the created objects section
+         *   showFiles {boolean} (default true) if true, show the list of files in the report
+         *   showHTML {boolean} (default true) if true, show raw HTML from the report, if it exists
+         *
+         * @returns the instantiated widget
+         */
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js:68]: A little missing commentary.

File: nbextensions/appCell2/widgets/tabs/resultsViewer.js [hunk 3]
```diff
@@ -15,119 +15,170 @@ define([
...
-                            hidden: true,
-                            type: 'default',
-                            classes: ['kb-panel-results'],
-                        }),
-                        div({ dataElement: 'report' }),
-                        div({ dataElement: 'next-steps' }),
-                    ]
-                );
-                container.innerHTML = layout;
+            return Promise.all([showResults(arg.jobState, arg.isParentJob), showNextApps()]);
```
Comment (briehl) [inline nbextensions/appCell2/widgets/tabs/resultsViewer.js:67]: Decomposed these into two separate Promissory functions.

File: nbextensions/appCell2/widgets/tabs/resultsViewer.js [hunk 4]
```diff
@@ -15,119 +15,170 @@ define([
...
-                }
+        /**
+         * Shows the results of a job run. This returns a Promise that resolves when
+         * the rendering is done.
+         * @param {Object} jobState - object describing the state of a job
+         * @param {boolean} isParentJob - true if this job is a parent of a batch job
+         * @returns
+         */
+        function showResults(jobState, isParentJob) {
+            // there's a couple of ways to find report parameters.
```
Comment (briehl) [inline nbextensions/appCell2/widgets/tabs/resultsViewer.js:78]: The guts of this are mostly unchanged from what used to be part of the `start` function. Just moved to `showResults` instead.

File: nbextensions/appCell2/widgets/tabs/resultsViewer.js [hunk 5]
```diff
@@ -15,119 +15,170 @@ define([
...
-                if (!reportRendered && DisplayUtil.verticalInViewport(reportElem)) {
-                    new KBaseReportView($(reportElem), reportParams);
-                    reportRendered = true;
-                }
-                if (reportRendered) {
-                    // Remove the scroll event listener if the report has been rendered.
-                    nbContainer.removeEventListener('scroll', lazyRenderReport);
+        function lazyRenderReport(reportParams) {
+            return Promise.try(() => {
+                const nbContainer = document.querySelector('#notebook-container');
```
Comment (briehl) [inline nbextensions/appCell2/widgets/tabs/resultsViewer.js:113]: Also mostly unchanged, just wrapped in a Promise. I'm not sure this is the best way, but it keeps the chains working right.

File: nbextensions/appCell2/widgets/tabs/resultsViewer.js [hunk 6]
```diff
@@ -15,119 +15,170 @@ define([
...
+            });
         }
 
+        /**
+         *
+         * @param {Object} params - parameters for the report view
+         */
         function renderReportView(params) {
-            reportParams = JSON.parse(JSON.stringify(params));
+            const reportParams = JSON.parse(JSON.stringify(params));
```
Comment (briehl) [inline nbextensions/appCell2/widgets/tabs/resultsViewer.js:143]: `reportParams` doesn't need to be global.

**Pull request merged**